server/controllers/eventController.js

```diff
@@ -0,0 +1,31 @@
+const Club = require("../models/clubModel");
+const Event = require("../models/clubEventModel");
+
+exports.createEvent = async (req, res) => {
+    try {
+        const { title, description, date, location, clubName } = req.body;
+
+        // Check if the club exists
+        const club = await Club.findOne({ name: clubName });
+        if (!club) {
+            return res.status(404).json({ message: "Club does not exist" });
```

Comment:
For this endpoint probably follow precedent set for error handling in other controllers: throw error, catch by message and output meaningful error/log.